GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -851,29 +851,38 @@ private void ShowLoading(bool showSpinner = true)
         }
 
         /// <summary>
-        ///  Indicates whether the revision grid can be refreshed, i.e. it is not currently being refreshed
-        ///  or it is not in a middle of reconfiguration process guarded by <see cref="SuspendRefreshRevisions"/>
+        ///  Indicates whether the revision grid can be refreshed,
+        ///  i.e. it is not in a middle of reconfiguration process guarded by <see cref="SuspendRefreshRevisions"/>
         ///  and <see cref="ResumeRefreshRevisions"/>.
         /// </summary>
-        private bool CanRefresh => !_isRefreshingRevisions && _updatingFilters == 0;
+        private bool CanRefresh => _suspendRefreshCounter == 0;
 
         #region PerformRefreshRevisions
 
         /// <summary>
         ///  Queries git for the new set of revisions and refreshes the grid.
         /// </summary>
         /// <exception cref="Exception"></exception>
-        /// <param name="forceRefresh">Refresh may be required as references may be changed.</param>
-        public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs = null, bool forceRefresh = false)
+        /// <param name="forceRefreshRefs">Refresh may be required as references may be changed.</param>
+        public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs = null, bool forceRefreshRefs = false)
         {
             ThreadHelper.AssertOnUIThread();
 
             if (!CanRefresh)
             {
-                Trace.WriteLine("Ignoring refresh as RefreshRevisions() is already running.");
+                Trace.WriteLine("Ignoring refresh as RefreshRevisions() is suspended.");
```

Comment:
[nit] For any trace I'd add a prefix that would help identifying the source, e.g.
```suggestion
                Trace.WriteLine("[PerformRefreshRevisions] Ignoring refresh as RefreshRevisions() is suspended.");
```